### PR TITLE
[WIP] u-boot: query customer OTP register in rpi3

### PR DESCRIPTION
**Pull request by acostach**
This patch adds support for querying the firmware
for one customer OTP register, as an alternative
for the case where bootmode register cannot be
inspected due to missing TAG implementation in firmware.

Since documentation does not seem to specify that TAG,
asked question in upstream whether it is available or not.

This patch is incomplete and should not be merged as-is.

Changelog-entry: u-boot: query customer OTP register in rpi3
Signed-off-by: Alexandru Costache <alexandru@balena.io>

### CODE REVIEW

**Comment by ZubairLK** [comment]
@floion interesting. there is a way from uboot to query otp registers.

perhaps @acostach can look into supporting usb boot
https://github.com/balena-os/balena-raspberrypi/issues/341
https://github.com/balena-os/meta-balena/issues/1575

tl;dr. in arch call, it was decided uboot should query otp bits if usb boot is enabled, then scan for usb devices as well if it is.

**Comment by floion** [comment]
@ZubairLK yes, that's the point of this PR

**Comment by lurch** [comment]
> Since documentation does not seem to specify that TAG, asked question in upstream whether it is available or not.

Just for reference: the relevant upstream issue appears to be https://github.com/raspberrypi/firmware/issues/1286
(and https://github.com/raspberrypi/firmware/issues/719 is also partially relevant).

**Comment by acostach** [comment]
Upstream response is there is no such tag to obtain the bootmode register using mailbox.

So we could check from u-boot if there's no sd-card with a BalenaOS image and consider uboot has been loaded from usb, or program any of the other 8 customer registers and check from there, etc.

**Comment by acostach** [comment]
Closing this PR since upstream developers confirmed firmware does not support querying for this specific data.
